src/python/bot/testcase_manager.py

```diff
@@ -820,17 +820,25 @@ def prepare_log_for_upload(symbolized_output, return_code):
   job_name = environment.get_value('JOB_NAME')
   components = revisions.get_component_list(app_revision, job_name)
   component_revisions = (
-      revisions.format_revision_list(components, use_html=False) or
-      'Not available.\n')
+    revisions.format_revision_list(components, use_html=False) or
+    "Not available.\n"
+  )
 
-  revisions_header = (
-      'Component revisions (build r{app_revision}):\n{component_revisions}\n'.
-      format(
-          app_revision=app_revision, component_revisions=component_revisions))
-  return_code_header = 'Return code: %s\n\n' % return_code
+  host_name = environment.get_value('HOSTNAME')
+  revisions_header = "Component revisions (build r{app_revision}):\
+  \n{component_revisions}\n{host_name}\n".format(
+    app_revision=app_revision,
+    component_revisions=component_revisions,
+    host_name=host_name,
+  )
+
+  if 'ANDROID' in environment.get_value('OS_OVERRIDE'):
+    revisions_header += environment.get_value('ANDROID_SERIAL') + "\n"
+
+  return_code_header = "Return code: %s\n\n" % return_code
 
   result = revisions_header + return_code_header + symbolized_output
-  return result.encode('utf-8')
+  return result.encode("utf-8")
```

Comment:
i think linter should complain, we use single-quote string all over code.

Reply:
my env is messed up. I cant seem to run the unit test cases or yapf. I did linting/formatting through external tools,

Reply:
I tried multiple times to clean the env and rebuild the dependencies but I kept getting the same issue when installing dependencies which is that python3-distutils can not be found, and when I try to install them, it fails saying they are already installed. Finally removing/adding resolved this issue.